src/codegen/sys/lib_.rs

```diff
@@ -90,6 +92,36 @@ fn get_extern_crate_string(library: &ExternalLibrary) -> String {
     )
 }
 
+fn include_custom_modules(w: &mut Write, env: &Env) -> Result<()> {
+    let modules = try!(find_modules(env));
+    if !modules.is_empty() {
+        try!(writeln!(w, ""));
+        for module in modules {
+            try!(writeln!(w, "pub mod {};", module));
```

Comment:
Maybe
```
mod foo;
pub use foo::*;
```
instead?

Reply:
You right, this give better flexibility.
@GuillaumeGomez You agree?

Reply:
Should there also be something to ignore autogenerated pieces then, so they can be manually implemented?

Reply:
[ignoring](https://github.com/gtk-rs/gir/blob/master/README.md#the-ffi-mode-toml-config) not enough?

Reply:
That's exactly what I thought of. For some reason I didn't notice that it is already implemented for FFI :)

Reply:
Ah but it can't be used for individual functions, or can it?

Reply:
Can same as in normal version as [code](https://github.com/gtk-rs/gir/blob/master/src/codegen/sys/functions.rs#L191) still here.

Reply:
> Maybe
>
> ```
> mod foo;
> pub use foo::*;
> ```
>
> instead?

I agree on it. :)